util/Migrator/DbScripts/2023-03-07_00_CipherKeyUpdate.sql

```diff
@@ -0,0 +1,438 @@
+-- Add Key Column
+IF COL_LENGTH('[dbo].[Cipher]', 'Key') IS NULL
+BEGIN
+    ALTER TABLE [dbo].[Cipher] ADD [Key] VARCHAR (MAX) NULL;
+END
+GO
+
+-- Add ForceKeyRotation
+IF COL_LENGTH('[dbo].[Cipher]', 'ForceKeyRotation') IS NULL
+BEGIN
+    ALTER TABLE [dbo].[Cipher] ADD [ForceKeyRotation] BIT DEFAULT (0) NOT NULL;
```

Comment:
⚠️ It's safer to add the column as nullable, set all values to 0, then alter it to be not null. I could be wrong about performance on this but I don't trust the default on column addition; worth quickly researching.

Reply:
I haven't investigated this myself but per https://contributing.bitwarden.com/contributing/code-style/sql#adding-a-column-to-a-table defaults should be far faster since the update will update every row. If we want to go this route this will need to be batched and run over a long period since Cipher has a lot of data.

Reply:
Great find -- I couldn't recall which was best.